packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -10,5 +147,101 @@ export interface TableControl<T extends StorageBrowserElements>
     | 'TableHeader'
     | 'TableRow'
   > {
-  (): React.JSX.Element;
+  <U extends ILocation>(props: TableControlProps<U>): React.JSX.Element;
+  SortIndeterminateIcon: T['Icon'];
+  SortAscendingIcon: T['Icon'];
+  SortDescendingIcon: T['Icon'];
 }
+
+const isSortableTableControlProps = <T extends ILocation>(
+  props: TableControlProps<T>
+): props is SortableTableControlProps<T> => {
+  return 'onSort' in props && 'sortState' in props;
+};
+
+const sortAriaLabel = {
+  none: 'not sorted',
+  ascending: 'sorted ascending',
+  descending: 'sorted descending',
+};
+
+export const TableControl: TableControl = <U extends ILocation>(
+  props: TableControlProps<U>
+) => {
+  const { data, ariaLabel } = props;
+  const { rows, columns } = data;
+
+  let onSort: (columnKey: Column<U>) => void;
+  let sortState: { key: keyof U; direction: SortDirection };
```

Comment:
The Table doesn't need to know anything abt sorting data, it's responsibility is to render the data provided to it